src/set.ts

```diff
@@ -157,11 +157,11 @@ export class SuperchargedSet<T> implements Iterable<T> {
   last (predicate?: (item: T, set: SuperchargedSet<T>) => unknown): T | undefined {
     if (predicate) {
       return SuperchargedSet.of(
-        this.toReverseArray()
-      ).find(predicate)
+        this.toArray()
+      ).reverse().find(predicate)

```

Comment:
Let’s shorten this statement because it contains redundant set creations. Calling `.reverse()` already creates a new set instance of the existing items.

You may shorten it to this:

```js
this.reverse().find(predicate)
```

Reply:
Yeah, sorry.